sklearn/metrics/pairwise.py

```diff
@@ -179,6 +193,19 @@ def euclidean_distances(X, Y=None, Y_norm_squared=None, squared=False,
     the distance matrix returned by this function may not be exactly
     symmetric as required by, e.g., ``scipy.spatial.distance`` functions.
 
+    Additionally, euclidean_distances() can also compute pairwise euclidean
```

Comment:
I suspect we should pull it out as a separate function, e.g. masked_euclidean_distances